### ISSUE TITLE
Custom cloud connector for AMQP / RabbitMQ

### ISSUE THREAD

**Issue by chicco785** [eclipse/kura#5042]
**Is your feature request related to a problem? Please describe.**

While MQTT is great in many scenarios, our solution adopt AMQP over RabbitMQ, thus instead of having to configure also MQTT (in rabbitmq or externally), it would be nice to have the possibility to create a RabbitMQ connector that allow to push messages to an exchange and consume them from a queue.

**Describe the solution you'd like**

The ability to develop a cloud connector that supports AMQP / RabbitMQ.

**Describe alternatives you've considered**

Using MQTT.

**Additional context**

N/A

**Comment by mattdibi** [comment]
Hello there @chicco785,

I don't think we're planning to support AMQP/RabbitMQ in the near future (@MMaiero correct me on this), but, if you feel brave enough, we have [a developer guide](https://eclipse.github.io/kura/docs-release-5.4/cloud-api/cloud-conn-dev-guide/#implement-cloudendpoint-and-cloudconnectionmanager) on how to implement a custom Cloud Connector that might be able to do what you need.

**Comment by chicco785** [comment]
> Hello there @chicco785,
> 
> I don't think we're planning to support AMQP/RabbitMQ in the near future (@MMaiero correct me on this), but, if you feel brave enough, we have [a developer guide](https://eclipse.github.io/kura/docs-release-5.4/cloud-api/cloud-conn-dev-guide/#implement-cloudendpoint-and-cloudconnectionmanager) on how to implement a custom Cloud Connector that might be able to do what you need.

Indeed I was more looking for guidance than actual implementation :)

Usage of amqp queues instead of mqtt may also be a way to achieve active-active deployments. I didn’t find any documentation on how to deploy kura in high availability, amqp queues may help, because messages are routed to a single consumer, so you can have two instances in parallel of kura with the same configuration and achieve h without message processing duplication.

**Comment by marcellorinaldo** [comment]
Hi @chicco785! I am currently writing a new Cloud Connection to be added as a Deployment Package in Kura.

I hope the following PR can be used as an example to start your development: https://github.com/eclipse/kura/pull/5059.

Keep the PR section monitored, I will add the implementation soon.

Best,
Marcello

**Comment by chicco785** [comment]
Thanks a lot @marcellorinaldo :)

**Comment by mattdibi** [comment]
Hello, just dropping by to highlight the addition of [this document](https://github.com/eclipse/kura/wiki/Cloud-Connectors-Implementation-Guidelines) in our Wiki that might be interesting for you (the document was originally created by @marcellorinaldo so I'm only stealing the credit here 🤫 ).

Additionally I will move this Issue into Discussion which seems to be a better fit for it.
